Application/EdFi.Ods.Extensions.TPDM/Artifacts/MsSql/Data/Security/1010-TPDM-ResourceClaims.sql

```diff
@@ -1,465 +1,6004 @@
-CROSS APPLY
-    (SELECT ActionId
-    FROM [dbo].[Actions]
-    WHERE ActionName IN ('Create', 'Read', 'Update', 'Delete')) AS ac
-WHERE ResourceName IN ('anonymizedStudent'
-, 'anonymizedStudentAssessments'
-, 'performanceMeasure'
-, 'performanceMeasureCourseAssociation'
-, 'professionalDevelopmentEvent'
-, 'recruitmentEvent'
-, 'staffTeacherPreparationProviderAssociation'
-, 'anonymizedStudentAssessment'
-, 'talentManagementGoal'
-, 'teacherCandidate'
-, 'teacherCandidateStudentGrowthMeasure'
-, 'teacherCandidateTeacherPreparationProviderAssociation'
-, 'anonymizedStudentAcademicRecord'
-, 'anonymizedStudentAssessmentCourseAssociation'
-, 'anonymizedStudentAssessmentSectionAssociation'
-, 'anonymizedStudentCourseAssociation'
-, 'anonymizedStudentCourseTranscript'
-, 'anonymizedStudentEducationOrganizationAssociation'
-, 'anonymizedStudentSectionAssociation'
-, 'applicant'
-, 'applicantProspectAssociation'
-, 'application'
-, 'applicationEvent'
-, 'completerAsStaffAssociation'
-, 'employmentEvent'
-, 'employmentSeparationEvent'
-, 'openStaffPositionEvent'
-, 'prospect'
-, 'staffApplicantAssociation'
-, 'fieldworkExperienceSectionAssociation'
-, 'staffProspectAssociation'
-, 'staffStudentGrowthMeasure'
-, 'staffStudentGrowthMeasureCourseAssociation'
-, 'staffStudentGrowthMeasureEducationOrganizationAssociation'
-, 'staffStudentGrowthMeasureSectionAssociation'
-, 'staffTeacherPreparationProviderProgramAssociation'
-, 'teacherCandidateTeacherPreparationProviderProgramAssociation'
-, 'teacherCandidateAcademicRecord'
-, 'teacherCandidateCourseTranscript'
-, 'teacherCandidateStaffAssociation'
-, 'teacherCandidateStudentGrowthMeasureCourseAssociation'
-, 'teacherCandidateStudentGrowthMeasureEducationOrganizationAssociation'
-, 'teacherCandidateStudentGrowthMeasureSectionAssociation'
-, 'teacherPreparationProviderProgram'
-, 'professionalDevelopmentEventAttendance'
-);
+BEGIN
+    DECLARE 
+        @applicationId AS INT,
+        @claimId AS INT,
+        @claimName AS nvarchar(max),
+        @parentResourceClaimId AS INT,
+        @existingParentResourceClaimId AS INT,
+        @claimSetId AS INT,
+        @claimSetName AS nvarchar(max),
+        @authorizationStrategyId AS INT,
+        @msg AS nvarchar(max),
+        @createActionId AS INT,
+        @readActionId AS INT,
+        @updateActionId AS INT,
+        @deleteActionId AS INT
+
+    DECLARE @claimIdStack AS TABLE (Id INT IDENTITY, ResourceClaimId INT)
+
+    SELECT @applicationId = ApplicationId
+    FROM [dbo].[Applications] WHERE ApplicationName = 'Ed-Fi ODS API';
+
+    SELECT @createActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Create';
+
+    SELECT @readActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Read';
+
+    SELECT @updateActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Update';
+
+    SELECT @deleteActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Delete';
+
+    BEGIN TRANSACTION
```

Comment:
This is here because during development I wanted a transaction around the changes so that I could re-extract the final XML metadata to do inspection before rolling back the transaction for the next iteration. I just never thought to remove it, but I will remove explicit transaction management from the final SQL Server scripts here (Postgres apparently does an implicit transaction around the whole pgsql block, from what I could tell).